ui/library/src/utils/ScrollDirectionDetector.ts

```diff
@@ -9,12 +9,31 @@ export default class ScrollDetector {
   _lastScrollDirection: Nullable<ScrollDirection>;
   _el: Element;
   _setScrollDirection: (scrollDirection: ScrollDirection) => any;
+  _lastScrollTopBeforeDirectionChange: number;
+  _scrollThreshold?: number;
+  _isScrollThresholdReachedInDirection: Nullable<ScrollDirection>;
+  _setScrollThresholdReachedInDirection?: (scrollDirection: Nullable<ScrollDirection>) => any;
+  _isAtTop: Nullable<boolean>;
+  _setIsAtTop: (isAtTop: boolean) => any;
 
-  constructor(el: Element, setScrollDirection: (scrollDirection: ScrollDirection) => any) {
+  constructor(
+    el: Element,
+    setScrollDirection: (scrollDirection: ScrollDirection) => any,
+    setIsAtTop: (isAtTop: boolean) => any,
+    setScrollThresholdReachedInDirection?: (scrollDirection: Nullable<ScrollDirection>) => any,
+    scrollThreshold?: number
+  ) {
     this._el = el;
     this._lastScrollTop = this._el.scrollTop;
     this._lastScrollDirection = null;
     this._setScrollDirection = setScrollDirection;
+    this._isAtTop = null;
+    this._setIsAtTop = setIsAtTop;
+    this._lastScrollTopBeforeDirectionChange = this._el.scrollTop;
```

Comment:
@ericmarsh995 what is the purpose of this?? Can u use `this._lastScrollTop`??

Reply:
`this._lastScrollTopBeforeDirectionChange` is needed for keeping track of total scroll amount since the direction changed. Without it, the scroll threshold wouldnt fire on total scroll amount and only on single large scrolls. 

`this._isAtTop` could probably be removed and use `this._lastScrollTop === 0` instead

Reply:
@ericmarsh995 is there any chance u can attack like a gif or vide of how your logic works with UI in PDF Component??

Reply:
Uploading Screen Recording 2022-07-25 at 12.30.30 PM.mov…

Reply:
https://user-images.githubusercontent.com/108751850/180859910-6190dc11-6d33-4dea-b831-76dfc609dfce.mov